### PR TITLE
linux, rootless: clamp oom_score_adj if it is too low

### PR DESCRIPTION
**Pull request by giuseppe**
when running rootless, if the specified oom_score_adj for the container process is lower than the current value, clamp it to the current value and print a warning.

Closes: https://github.com/containers/podman/issues/19829

#### Does this PR introduce a user-facing change?

```release-note
When running rootless, if the oom-score-adj value is too low, then it is clamped to the current value
```

### CODE REVIEW

**Review by vrothberg — APPROVED** [review]
Code LGTM

Can you add a note to the docs as well?

**Comment by giuseppe** [comment]
> Can you add a note to the docs as well?

sure, added a note to the docs

**Review by flouthoc — APPROVED** [review]
LGTM
/hold 
/lgtm

**Comment by giuseppe** [comment]
needs again the lgtm label

**Comment by Luap99** [comment]
Late comment but doesn't this run into the same problem similar to https://github.com/containers/podman/pull/18721?

You know set the limit on container creation permanently. Which means a long living container cannot be started if the oom_score_adj limit for the current user is set higher.
Instead we should check this value on startup time based on the real current value and not the calue at creation time.

**Comment by giuseppe** [comment]
> Late comment but doesn't this run into the same problem similar to #18721?
> 
> You know set the limit on container creation permanently. Which means a long living container cannot be started if the oom_score_adj limit for the current user is set higher. Instead we should check this value on startup time based on the real current value and not the calue at creation time.

That is a good idea, taking a look right now

**Comment by giuseppe** [comment]
@Luap99 PR here: https://github.com/containers/podman/pull/19927

**Comment by snazy** [comment]
Maybe not the right place to ask, but could you push a new 4.6.3 patch release with this fix?
I'm kind of locked into [SuSE's unstable repo](https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/unstable/xUbuntu_22.04/amd64/) and the workaround to downgrade crun doesn't work there (switching to the "stable" one doesn't work for other reasons).

**Pull request merged**
